rbac/management/principal/it_service.py

```diff
@@ -303,11 +303,33 @@ def get_service_accounts(self, user: User, options: dict[str, Any] = {}) -> Tupl
         offset = options.get("offset")
         limit = options.get("limit")
         limit_offset_validation(offset, limit)
-
+        # Service account filtering query parameters
+        name = options.get("name")
+        owner = options.get("owner")
+        description = options.get("description")
+        filtered_service_accounts = []
+        sa_query_passed = name or owner or description
         count = len(service_accounts)
         # flake8 ignore E203 = Whitespace before ':' -> false positive https://github.com/PyCQA/pycodestyle/issues/373
         service_accounts = service_accounts[offset : offset + limit]  # type: ignore # noqa: E203
-
+        # If any one service account filter parameter is provided extract & return the specific service accounts
+        if sa_query_passed:
+            for sa in service_accounts:
+                sa_description = str(sa.get("description"))
+                if (
+                    (name and sa.get("name") == name)
+                    or (owner and sa.get("owner") == owner)
+                    or (description and description in sa_description)
+                ):
+                    filtered_service_accounts.append(sa)
+                    count = len(filtered_service_accounts)
+                    service_accounts = filtered_service_accounts
+        else:
+            # If any order_by parameter is passed then sort the service accounts by that field either asc or desc
+            if order_by in ["-time_created", "-name", "-description", "-clientId", "-owner"]:
+                service_accounts.sort(reverse=True, key=lambda sa: sa.get(order_by[1:], ""))
```

Comment:
Maybe it's a good idea to have a case-insensitive ordering:
```suggestion
                service_accounts.sort(reverse=True, key=lambda sa: sa.get(order_by[1:], "").casefold())
```